### DISCUSSION TITLE
Replace one color with another.

### DISCUSSION THREAD

**Discussion by jazzzsoftware** [mono/SkiaSharp#1738] [Q&A] [answered]
Hi,

I start with an image of a blue t-shirt on a white background. I want to replace the white with the transparent color.

How can I achieve this?

Thanks

Mark.

**Comment by jazzzsoftware** [comment, marked as answer]
I solved this by using ImageSharp. 

I have a Skia solution [here](https://groups.google.com/g/skia-discuss/c/RFOcoin5YsM) but I could not get it to work in SkiaSharp.

I investigated other libraries and ImageSharp did the job. Here is the [solution](https://github.com/SixLabors/ImageSharp.Drawing/discussions/147).

I would prefer to just use SkiaSharp so please post the SkiaSharp solution if you get it to work :)

**Reply by mattleibow** [reply]
You can use the SKRuntimeEffect as here

https://github.com/mono/SkiaSharp/blob/main/tests/Tests/SKRuntimeEffectTest.cs#L728

The one issue right now is this is just supported on the GPU. However this is just a current build limitation and if it is needed that I can have a look at getting CPU support compiled in.

Might be worth using a GPU view for these operations for performance. But skia is just so fast I think most people won't even notice.

**Reply by mattleibow** [reply]
There might be a way to do this work colour filters. You could draw the shirt onto a transparent canvas, but with a blend mode of additive or similar. Then the white will not be dawn...

**Reply by jazzzsoftware** [reply]
Thanks for your suggestions. Do you know how I could use a color filter? Do you have a code example?
